src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -219,9 +313,149 @@ repl_req_ptr_t RaftReplDev::follower_create_req(repl_key const& rkey, sisl::blob
     rreq->local_blkid = do_alloc_blk(data_size, m_listener->get_blk_alloc_hints(user_header, data_size));
     rreq->state.fetch_or(uint32_cast(repl_req_state_t::BLK_ALLOCATED));
 
+    RD_LOG(INFO, "in follower_create_req: rreq={}, addr={}", rreq->to_compact_string(),
+           reinterpret_cast< uintptr_t >(rreq.get()));
     return rreq;
 }
 
+void RaftReplDev::check_and_fetch_remote_data(std::vector< repl_req_ptr_t >* rreqs) {
+    // Pop any entries that are already completed - from the entries list as well as from map
+    rreqs->erase(std::remove_if(
+                     rreqs->begin(), rreqs->end(),
+                     [this](repl_req_ptr_t const& rreq) {
+                         if (rreq == nullptr) { return true; }
+
+                         if (rreq->state.load() & uint32_cast(repl_req_state_t::DATA_WRITTEN)) {
+                             m_repl_key_req_map.erase(rreq->rkey); // Remove=Pop from map as well, since it is completed
+                             RD_LOG(INFO,
+                                    "Raft Channel: Data write completed and blkid mapped, removing from map: rreq=[{}]",
+                                    rreq->to_compact_string());
+                             return true; // Remove from the pending list
+                         } else {
+                             return false;
+                         }
+                     }),
+                 rreqs->end());
+
+    if (rreqs->size()) {
+        // Some data not completed yet, let's fetch from remote;
+        fetch_data_from_remote(rreqs);
```

Comment:
now, we do check_and_fetch_remote_data when after the log append. why can not we delay this action until we need it or using some periodical check thread to do? maybe the push_data requset is on the way

Reply:
As indicated above, this the exactly what this PR is trying to do. e.g. we need the timer to wait for a configurable time of seconds before we fetch from remote.